addons/sourcemod/scripting/AntiBhopCheat_Discord.sp

```diff
@@ -0,0 +1,212 @@
+	g_cvWebhookRetry = CreateConVar("sm_antibhopcheat_discord_webhook_retry", "3", "Number of retries if webhook fails.", FCVAR_PROTECTED);
+	g_cvAvatar = CreateConVar("sm_antibhopcheat_discord_avatar", "https://avatars.githubusercontent.com/u/110772618?s=200&v=4", "URL to Avatar image.");
+	g_cvChannelType = CreateConVar("sm_antibhopcheat_discord_channel_type", "0", "Type of your channel: (1 = Thread, 0 = Classic Text channel");
+
+	/* Thread config */
+	g_cvThreadName = CreateConVar("sm_antibhopcheat_discord_threadname", "AntiBhopCheat - New detection", "The Thread Name of your Discord forums. (If not empty, will create a new thread)", FCVAR_PROTECTED);
+	g_cvThreadID = CreateConVar("sm_antibhopcheat_discord_threadid", "0", "If thread_id is provided, the message will send in that thread.", FCVAR_PROTECTED);
+	AutoExecConfig(true);
+}
+
+public void OnAllPluginsLoaded()
+{
+	g_Plugin_SourceBans = LibraryExists("sourcebans++");
+	g_Plugin_ExtDiscord = LibraryExists("ExtendedDiscord");
+}
+
+public void OnLibraryAdded(const char[] sName)
+{
+	if (strcmp(sName, "sourcebans++", false) == 0)
+		g_Plugin_SourceBans = true;
+	if (strcmp(sName, "ExtendedDiscord", false) == 0)
+		g_Plugin_ExtDiscord = true;
+}
+
+public void OnLibraryRemoved(const char[] sName)
+{
+	if (strcmp(sName, "sourcebans++", false) == 0)
+		g_Plugin_SourceBans = false;
+	if (strcmp(sName, "ExtendedDiscord", false) == 0)
+		g_Plugin_ExtDiscord = false;
+}
+
+public void OnMapInit(const char[] mapName)
+{
+	FormatEx(g_sMap, sizeof(g_sMap), mapName);
+}
+
+public void AntiBhopCheat_OnClientDetected(int client, char[] sReason, char[] sStats)
+{
+	char sWebhookURL[WEBHOOK_URL_MAX_SIZE];
+	g_cvWebhook.GetString(sWebhookURL, sizeof sWebhookURL);
+	if (!sWebhookURL[0]) {
+		LogError("[%s] No webhook found or specified.", PLUGIN_NAME);
+		return;
+	}
+
+	char sAuth[32];
+	GetClientAuthId(client, AuthId_Steam2, sAuth, sizeof(sAuth), false);
+
+	char sPlayer[256];
+	if (g_Plugin_SourceBans) {
+		int iClientBans = 0;
+		int iClientComms = 0;
+
+	#if defined _sourcebanschecker_included
+		iClientBans = SBPP_CheckerGetClientsBans(client);
+		iClientComms = SBPP_CheckerGetClientsComms(client);
+	#endif
+
+		FormatEx(sPlayer, sizeof(sPlayer), "%N (%d bans - %d comms) [%s] %s.", client, iClientBans, iClientComms, sAuth, sReason);
+	} else {
+		FormatEx(sPlayer, sizeof(sPlayer), "%N [%s] %s.", client, sAuth, sReason);		
+	}
+
+	char sTime[64];
+	int iTime = GetTime();
+	FormatTime(sTime, sizeof(sTime), "Date : %d/%m/%Y @ %H:%M:%S", iTime);
+
+	char sCount[32];
+	int iMaxPlayers = MaxClients;
+	int iConnected = GetPlayerCount(g_cvCountBots.BoolValue);
+	FormatEx(sCount, sizeof(sCount), "Players : %d/%d", iConnected, iMaxPlayers);
+
+	char sHeader[512], sMessage[2000];
+	FormatEx(sHeader, sizeof(sHeader), "%s \nCurrent map : %s \n%s \n%s", sPlayer, g_sMap, sTime, sCount);
+
+	// Discord character limit is 2000 (discord msg + stats)
+	if (strlen(sHeader) + strlen(sStats) < 1994)
```

Comment:
hardcoded value, put it somewhere as a define (maybe in the core discord include file)